src/components/StylingDrawer/SldStylingPanel/index.tsx

```diff
@@ -0,0 +1,109 @@
+import React, {
+  useEffect, useState, useCallback
+} from 'react';
+
+import {
+  CardStyle, CardStyleProps
+} from 'geostyler/dist/Component/CardStyle/CardStyle';
+import SLDParser from 'geostyler-sld-parser';
+import { Style as GsStyle } from 'geostyler-style';
+import Layer from 'ol/layer/Layer';
+import {
+  ImageWMS, TileWMS
+} from 'ol/source';
+
+import { Logger } from '@terrestris/base-util';
+import { MapUtil } from '@terrestris/ol-util';
+import { useMap } from '@terrestris/react-util/dist/Hooks/useMap/useMap';
+
+import useAppSelector from '../../../hooks/useAppSelector';
+import useSHOGunAPIClient from '../../../hooks/useSHOGunAPIClient';
+import {
+  fetchGeoserverStyle, fetchWorkspaceFromGetCapabilities, getLayerUrl
+} from '../../../utils/geoserverUtils';
+
+export type SldStylingPanelProps = CardStyleProps;
+
+const SldStylingPanel: React.FC<SldStylingPanelProps> = (props): JSX.Element => {
+  const [style, setStyle] = useState<GsStyle | undefined>();
+  const layerUid = useAppSelector(state => state.stylingDrawerLayerUid);
+  const map = useMap();
+  const client = useSHOGunAPIClient();
+
+  const getStyle = useCallback(async () => {
+    if (!map || !layerUid) {
+      return;
+    }
+
+    const layer = MapUtil.getLayerByOlUid(map, layerUid) as Layer;
+    if (!layer) {
+      return;
+    }
+
+    const source = layer.getSource();
+    if (!(source instanceof ImageWMS || source instanceof TileWMS)) {
+      return;
+    }
+
+    const layerUrl = getLayerUrl(source);
+    if (!layerUrl) {
+      return;
+    }
+
+    try {
+      const workspaceInfo = await fetchWorkspaceFromGetCapabilities(layerUrl, client);
```

Comment:
Just wondering why this is needed. Usually the workspace is already party of the layer name, e.g. `MYWORKSPACE:MYLAYER`. This can probably enhanced by checking if the name contains the workspace already and if not getting it from the capabilities.